### DISCUSSION TITLE
common_neighbors for directed graphs

### DISCUSSION THREAD

**Discussion by dmaniloff** [networkx/networkx#5140] [Q&A] [answered]
Hello networkx community! Thank you for networkx 🙏 .

Question: why is `common_neighbors` not implemented for directed graphs? 

It seems that `(w for w in G[u] if w in G[v] and w not in (u, v))` plus the checks for membership should be valid for both undirected and directed graphs:
https://github.com/networkx/networkx/blob/486c89bb710b13f4c6dcd6751c260b678d37b322/networkx/classes/function.py#L925

I am looking to calculate `jaccard_coefficient` and `adamic_adar_index` and `common_neighbor_centrality` (and others) for a directed graph and they all depend on `common_neighbors`.

Let me know if this is possible and I will gladly submit a PR. Thanks!

**Comment by rossbar** [comment]
Good question! At face value I agree with your assessment - I don't see any obvious reason why this should be limited to undirected graphs. I checked the history a bit and it looks like this was last modified when link_prediction was introduced in #1147 and #1148. I didn't see any justification in the discussions there for why this was limited to undirected graphs, but maybe I just missed it.

**Comment by dschult** [comment, marked as answer]
I agree... If you can supply a PR for it, please do. 
Thanks!
